backend/shared/shared/azure_adfs/urls.py

```diff
@@ -1,8 +1,39 @@
 from django.urls import include, path, re_path
+from django_auth_adfs.views import OAuth2LoginView, OAuth2LogoutView
 
 from shared.azure_adfs.views import HelsinkiOAuth2CallbackView
+from shared.common.views import MockEnabledProxyView
+
+from .mock_views import (
+    MockOAuth2CallbackView,
+    MockOAuth2LoginView,
+    MockOAuth2LogoutView,
+)
+
+app_name = "django_auth_adfs"
 
 urlpatterns = [
-    re_path(r"^callback$", HelsinkiOAuth2CallbackView.as_view(), name="callback"),
+    re_path(
+        r"^callback$",
+        MockEnabledProxyView(
+            real_view_class=HelsinkiOAuth2CallbackView,
+            mock_view_class=MockOAuth2CallbackView,
+        ),
+        name="callback",
+    ),
+    re_path(
+        r"^login$",
+        MockEnabledProxyView(
+            real_view_class=OAuth2LoginView, mock_view_class=MockOAuth2LoginView
+        ),
+        name="login",
+    ),
+    re_path(
+        r"^logout$",
+        MockEnabledProxyView(
+            real_view_class=OAuth2LogoutView, mock_view_class=MockOAuth2LogoutView
+        ),
+        name="logout",
+    ),
     path("", include("django_auth_adfs.urls")),
```

Comment:
Where did the login and logout endpoints come from before this change? Did they come from `path("", include("django_auth_adfs.urls"))`? If they did then are you sure the login and logout mocks are not being overridden by values from `path("", include("django_auth_adfs.urls"))`?

Reply:
Yes, they came from there `django_auth_adfs`. URLs are not overridden, as this works on the "first that matches" principle.